frappe/email/doctype/notification/notification.py

```diff
@@ -282,6 +282,9 @@ def send_an_email(self, doc, context):
 				bcc=bcc,
 				communication_type="Automated Message",
 			).get("name")
+			# set the outgoing email account because we did in fact send it via sendmail above
+			comm = frappe.get_doc("Communication", communication)
+			comm.get_outgoing_email_account()
```

Comment:
I don't get what this PR is trying to do :thinking: 

>  set the outgoing email account because **we did in fact send it via sendmail above**

It says `send_email=False`, so it's not sending?

Reply:
Ah, it is already sent by `.sendmail`, so it should apear in the communication log as "Sent", but without this, it doesn't.

Reply:
The `get_outgoing_email_account` is an implicit setter. That's probably what confuses a bit.

Reply:
It's setting what? 

I don't see any unsent automated messages on our servers. 

![image](https://github.com/frappe/frappe/assets/9079960/156906dc-d60d-4c40-b066-50d3165810df)

Reply:
Hm, that datum disagrees with my recollection of the observation I had. :thinking:

Reply:
Ah sorry, we have the case of an outdated recollection...

Reply:
This PR is after: `self.db_set("email_account", self._outgoing_email_account.name)` so that it shows when selecting the corresponding email account.

```python
	def get_outgoing_email_account(self):
		if not hasattr(self, "_outgoing_email_account"):
			if self.email_account:
				self._outgoing_email_account = EmailAccount.find(self.email_account)
			else:
				self._outgoing_email_account = EmailAccount.find_outgoing(
					match_by_email=self.sender_mailid, match_by_doctype=self.reference_doctype
				)

				if self.sent_or_received == "Sent" and self._outgoing_email_account:
					if frappe.db.exists("Email Account", self._outgoing_email_account.name):
						self.db_set("email_account", self._outgoing_email_account.name)

		return self._outgoing_email_account
```